app/Services/InviteService.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace App\Services;
+
+use App\Mail\InviteCreated;
+use App\Mail\Reinvite;
+use App\Models\InviteToken;
+use App\Models\UserRole;
+use Carbon\Carbon;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Mail;
+use Illuminate\Support\Str;
+
+
+class InviteService
+{
+public function generateToken():string{
+    $random=Str::random(60);
+    $time=Carbon::now();
+    return $random.$time->toDateTimeLocalString();
+}
+    public function invite($name, $email, $role_id, $vendor_employee_id): bool
+    {
+        $token = $this->generateToken();
+        $url=config('=.url') . '/register' .$token  . '?email=' .$email .'?name=' .$name . '?role_id=' .$role_id;
+
+        $employeeVendor = InviteToken::create([
+            'name' => $name,
+            'email' => $email,
+            'role_id' => $role_id,
+            'token' => $token,
+            'tokenExpires' => Carbon::now()->addDays(5),
+            'invitedUserId' => $vendor_employee_id
+        ]);
+        if (!$employeeVendor) return false;
```

Comment:
@Babita3331 If for some reason, a new record is not created in the database, an error will be thrown directly. However, you do have to handle it in the global handler, or create your own exception class